### ISSUE TITLE
Auto deploy not picking up machines that fire events while an auto deploy is in progress

### ISSUE THREAD

**Issue by tothegills** [OctopusDeploy/Issues#3013]
If a machine fires events while an automatic deployment is in progress it will not get picked up for the automatic deployment of that release.

Machines that fire events during the initial deployment work as expected.

In this example, web-server_0.0.5 came online while the deployment to web-server_0.0.4 was in progress.  Consequently it was not deployed to.
![image](https://cloud.githubusercontent.com/assets/171197/21205498/454010ee-c2a8-11e6-8bc3-36f3b108dbaa.png)

See http://help.octopusdeploy.com/discussions/problems/49007

Caused by the fix to https://github.com/OctopusDeploy/Issues/issues/2860

**Comment by tothegills** [comment]
Did not actually reproduce the issue, what I was seeing was my machines bumping up against the licence limit and preventing a deployment.  Still investigating.

**Comment by thebigmoosey** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Auto deploy not picking up machines that fire events while an auto deploy is in progress](https://trello.com/c/N8m1eABc/18-auto-deploy-not-picking-up-machines-that-fire-events-while-an-auto-deploy-is-in-progress)

**Comment by thebigmoosey** [comment]
Ok wow, we managed to reproduce this finally. It's a timing issue when you have long-running deployments blocking auto-deploy, between (1) when auto-deploy does machine lookups and (2) when Octopus is able to re-queue the deployment task (logging a `DeploymentStarted` event). There can be machines coming online during the time between (1) and (2) and these machines may get skipped/missed when auto-deploy comes around to picking up machines that were blocked during a long-running deployment.

See this dev task log for an example of the behaviour (note `tentacle-11` does not get deployed to):
<img width="1732" alt="screen shot 2017-01-16 at 10 51 04 am" src="https://cloud.githubusercontent.com/assets/819605/21969071/e01dfea8-dbe6-11e6-9067-ed696c53398d.png">

This wasn't originally an issue, but as mentioned above, when we fixed the performance problems around [this issue](https://github.com/OctopusDeploy/Issues/issues/2860) we introduced this new edge case.

Looking into a fix now.
